source/tests/test_model_format_utils.py

```diff
@@ -108,6 +109,61 @@ def test_embedding_net(self):
             np.testing.assert_allclose(en0.call(inp), en1.call(inp))
 
 
+class TestNetworks(unittest.TestCase):
+    def setUp(self) -> None:
+        w = np.full((2, 3), 3.0)
+        b = np.full((3,), 4.0)
+        self.network = {
+            "layers": [
+                {
+                    "activation_function": "tanh",
+                    "resnet": True,
+                    "@variables": {"w": w, "b": b},
+                },
+                {
+                    "activation_function": "tanh",
+                    "resnet": True,
+                    "@variables": {"w": w, "b": b},
+                },
+            ],
+        }
+
+    def test_two_dim(self):
+        networks = Networks(ndim=2)
+        networks[(0, 0)] = self.network
+        networks[(0, 1)] = self.network
```

Comment:
should we assume the completeness of the networks? in case ndim==2 and ntypes==2, the networks (0, 0), (0, 1), (1, 0) and (1, 1) should be provided.